tests/integration/states/test_idem.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the idem state
+"""
+import tempfile
+from contextlib import contextmanager
+
+import salt.utils.idem as idem
+import salt.utils.path
+from tests.support.case import ModuleCase
+from tests.support.mixins import SaltReturnAssertsMixin
+from tests.support.unit import skipIf
+
+SLS_SUCCEED_WITHOUT_CHANGES = """
+state_name:
+  test.succeed_without_changes:
+    - name: idem_test
+    - foo: bar
+"""
+
+
+@skipIf(not idem.HAS_POP[0], str(idem.HAS_POP[1]))
+@skipIf(not salt.utils.path.which("idem"), "idem is not installed")
+@contextmanager
```

Comment:
Why do you need the context manager decorator?

Reply:
The state being tested uses `__context__` and that doesn't seem to get generated properly without the contextmanager decorator

Reply:
Huh?! Really?! ..... Ok....

I'm going to be a jerk down the line and I'll ask you to rewrite this for PyTest. Or I'll do it myself, but don't worry about that now.